src/backend/access/brin/brin.c

```diff
@@ -1204,7 +1205,8 @@ brin_summarize_range_internal(PG_FUNCTION_ARGS)
 
 	/* Must be a BRIN index */
 	if (indexRel->rd_rel->relkind != RELKIND_INDEX ||
-		indexRel->rd_rel->relam != BRIN_AM_OID)
```

Comment:
use this function IsIndexAccessMethod(Oid relam, Oid IndexAccessMethod) instead ?
indexRel->rd_rel->relam != BRIN_AM_OID --> IsIndexAccessMethod(indexRel->rd_rel->relam, BRIN_AM_OID)

Reply:
+1

Reply:
Thank for @HuSen8891 @avamingli. I had made a change, you can review then again.